java-extensions/jni-connector/src/main/java/com/starrocks/jni/connector/OffHeapTable.java

```diff
@@ -149,66 +112,54 @@ public int getNumRows() {
         }
         return meta.valuesNativeAddress();
     }
 
-    /**
-     * For test only
-     */
-    public void show(int limit) {
+    public void print(int rowLimit) {
         StringBuilder sb = new StringBuilder();
-        System.out.println("numRows = " + numRows);
-        for (int i = 0; i < limit && i < numRows; i++) {
-            for (int fieldId = 0; fieldId < types.length; fieldId++) {
-                OffHeapColumnVector column = vectors[fieldId];
-                if (column.isNullAt(i)) {
-                    sb.append("NULL").append(", ");
-                    continue;
-                }
-                OffHeapColumnVector.OffHeapColumnType type = types[fieldId];
-                switch (type) {
-                    case BOOLEAN:
-                        sb.append(column.getBoolean(i)).append(", ");
-                        break;
-                    case SHORT:
-                        sb.append(column.getShort(i)).append(", ");
-                        break;
-                    case INT:
-                        sb.append(column.getInt(i)).append(", ");
-                        break;
-                    case FLOAT:
-                        sb.append(column.getFloat(i)).append(", ");
-                        break;
-                    case LONG:
-                        sb.append(column.getLong(i)).append(", ");
-                        break;
-                    case DOUBLE:
-                        sb.append(column.getDouble(i)).append(", ");
-                        break;
-                    case STRING:
-                    case DATE:
-                    case DECIMAL:
-                        sb.append(column.getUTF8String(i)).append(", ");
-                        break;
-                    default:
-                        throw new RuntimeException("Unhandled " + type);
+        sb.append("OffHeapTable: numRows = " + numRows + "\n");
+        for (int i = 0; i < rowLimit && i < numRows; i++) {
+            sb.append("row" + i + ": [");
+            for (int j = 0; j < vectors.length; j++) {
+                if (j != 0) {
+                    sb.append(',');
                 }
+                sb.append(fields[j]).append(':');
+                vectors[j].dump(sb, i);
             }
-            sb.append("\n");
+            sb.append("]\n");
+        }
+        System.out.print(sb);
+    }
+
+    public static class MetaChecker {
+        private OffHeapColumnVector meta;
+        int offset;
+
+        public MetaChecker(OffHeapColumnVector meta, int offset) {
+            this.meta = meta;
+            this.offset = offset;
+        }
+
+        public void check(long expected) {
+            if (meta.getLong(offset) != expected) {
+                throw new RuntimeException(
+                        "meta check failed at offset: " + offset + ", act = " + meta.getLong(offset) + ", exp = " +
+                                expected);
+            }
+            offset += 1;
+        }
+    }
```

Comment:
It is recommended to add a `resetOffset()` method to avoid repeated calls by others accidentally and call reset at the end of `checkMeta()` for code robustness

Reply:
Actually I think there is no need to call this function. Maybe we can add it later when we need it.